src/test/java/qna/domain/QuestionTest.java

```diff
@@ -1,6 +1,49 @@
 package qna.domain;
 
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import qna.CannotDeleteException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
 public class QuestionTest {
     public static final Question Q1 = new Question("title1", "contents1").writeBy(UserTest.JAVAJIGI);
     public static final Question Q2 = new Question("title2", "contents2").writeBy(UserTest.SANJIGI);
+
+    @Test
+    @DisplayName("현재 유저가 작성자이면 질문을 삭제할 수 있다.")
+    void deleteQuestionTest() throws CannotDeleteException {
+        Answer answer = new Answer(11L, UserTest.JAVAJIGI, QuestionTest.Q1, "Answers Contents1");
+        Q1.addAnswer(answer);
+
+        List<DeleteHistory> deleteHistories = Q1.deleteQuestionAndAnswers(UserTest.JAVAJIGI);
+        assertAll(
+                () -> assertThat(deleteHistories).hasSize(2),
+                () -> assertThat(Q1.isDeleted()).isTrue(),
+                () -> assertThat(Q1.getAnswers().get(0).isDeleted()).isTrue()
```

Comment:
```suggestion
                () -> assertThat(answer.isDeleted()).isTrue()
```
지역 변수를 활용해도 좋을 것 같아요 :)

Reply:
말씀하신대로 `mappedAnswer` 등의 네임으로 지역변수 추출이 더 나을 것 같습니다!
다만 해당 테스트에서 `.get(0)` 을 하면 확실히 하나의 answer 이 나온다는 것을 알고 있어서  `.get(0)` 을 사용했는데, 테스트 리팩토링 이라던가 수정이 있을 경우를 대비해서 Optional 혹은 
`
List<Answer> answers = Q1.getAnswers();
answers.forEach(answer1 -> assertThat(answer1.isDeleted()).isTrue());
`

이런식으로 리스트 전체에 대한 테스트를 하는게 나을지 고민했었습니다!
혹시 리뷰어님께서는 이럴 경우에 어떤식으로 테스트를 작성하시는지 여쭤봐도 될까요!

Reply:
현재 테스트에서 단순히 삭제 여부만 확인하기 때문에 
Q1에 등록되어 있는 Answer가 많아진다면
저도 말씀해주신 것 처럼 `answers.forEach(answer1 -> assertThat(answer1.isDeleted()).isTrue());` 방식을 사용할 것 같아요 👍👍